wyvern/core/http.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+import logging
+
+import aiohttp
+
+from wyvern.exceptions import WyvernError
+
+logger = logging.getLogger(__name__)
+DEFAULT_REQUEST_TIMEOUT = 60
+timeout = aiohttp.ClientTimeout(total=DEFAULT_REQUEST_TIMEOUT)
+
+
+class AiohttpClientWrapper:
+
+    async_client = None
+
+    def start(self):
+        """Instantiate the client. Call from the FastAPI startup hook."""
+        self.async_client = aiohttp.ClientSession(timeout=timeout)
+
+    async def stop(self):
+        """Gracefully shutdown. Call from FastAPI shutdown hook."""
+        if not self.async_client:
+            return
+        if self.async_client and not self.async_client.closed:
+            await self.async_client.close()
+        self.async_client = None
+
+    def __call__(self):
+        """Calling the instantiated AiohttpClientWrapper returns the wrapped singleton."""
+        # Ensure we don't use it if not started / running
+        if self.async_client is None:
+            raise WyvernError("AiohttpClientWrapper not started")
+
+        return self.async_client
+
+
+aiohttp_client = AiohttpClientWrapper()
```

Comment:
Should we just call this http_client? Or does it make sense to also keep the aio

Reply:
yeah, good idea. i'll update

Reply:
actually I'll keep it aiohttp_client for now.